### PR TITLE
Add analytics to live radio pages

### LINKED ISSUE #2964: Live Radio Analytics
**Is your feature request related to a problem? Please describe.**
Ability to track live radio page metrics with ATI Analytics

Continue to track live radio page metrics with ATI and not lose any data when we make the switch to Simorgh pages

 Metrics will be captured on WS_NEWS_LANGUAGES_TEST (site ID - 598343) until we release to production at which point we will capture analytics through WS_NEWS_LANGUAGES 

**Testing notes**
[Tester to complete]

**Additional context**
WSMEDIA-26

### PR DESCRIPTION
**Pull request by ryanmccombe**
Resolves https://github.com/bbc/simorgh/issues/2964  https://github.com/bbc/simorgh/issues/3136

**Overall change:** _Add analytics tracking to Live Radio pages._

**Code changes:**

- _Add additional function to parse live radio data into ATI URL._

**Test notes:**

- Needs testing - can use the chrome / FF plugin to check the correct params are being passed to ATI: https://chrome.google.com/webstore/detail/at-internet-tag-inspector/epdfbeoiphkaeapcohmilhmpdeilgnok?hl=en

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: src/app/containers/ATIAnalytics/index.jsx
Comment (dr3) [inline src/app/containers/ATIAnalytics/index.jsx]: This object has the same keys as the array on line 15, can you use `Object.keys(pageviewParams)` to save duplication?

File: src/app/containers/ATIAnalytics/params/media/index.jsx
Comment (dr3) [inline src/app/containers/ATIAnalytics/params/media/index.jsx]: Can this use `pathOr`?

Comment (jroebu14) [reply src/app/containers/ATIAnalytics/params/media/index.jsx]: or maybe just `path`?

File: src/app/containers/ATIAnalytics/params/media/index.test.jsx
Comment (jroebu14) [inline src/app/containers/ATIAnalytics/params/media/index.test.jsx]: should we perhaps use `testing-library/react` library instead? I think there was chat of exclusively using `testing-library/react` and eventually removing `react-test-renderer`

File: src/app/containers/ATIAnalytics/index.jsx
Comment (ryanmccombe) [reply src/app/containers/ATIAnalytics/index.jsx]: @dr3 do you mean like this?

Comment (ryanmccombe) [reply src/app/containers/ATIAnalytics/index.jsx]: @dr3 updated it - do you mean like this?

File: src/app/containers/ATIAnalytics/index.jsx
Comment (jroebu14) [inline src/app/containers/ATIAnalytics/index.jsx]: probably wouldn't need the above comment if you did something like

```js
const pageTypeNotRecognised = !Object.keys(pageTypeHandlers).includes(pageType)

if (pageTypeNotRecognised) {
```

File: src/app/containers/ATIAnalytics/index.jsx
Comment (jroebu14) [inline src/app/containers/ATIAnalytics/index.jsx]: I also think the comment above wouldn't be needed if the functions were named better e.g. `getArticleAtiParams`. The capitalisation also suggests it's a React component

Comment (ryanmccombe) [reply src/app/containers/ATIAnalytics/index.jsx]: @jroebu14 updated

These handlers actually are react components.  Confused me at first as they just return a string, but it needs to be a component in order to use `React.context`

File: data/indonesia/bbc_indonesian_radio/liveradio.json
Comment (pjlee11) [inline data/indonesia/bbc_indonesian_radio/liveradio.json:7]: `pageIndentifier` seems to map to `counterName` in the other types of Ares payloads (Indexes and CPS pages) is there a reason we wouldn't want this to remain consistent?

File: data/indonesia/bbc_indonesian_radio/liveradio.json
Comment (pjlee11) [inline data/indonesia/bbc_indonesian_radio/liveradio.json:8]: Have we made this field up? 

It seems that is is being used to populate the `atiPageViewParams` which for both the index and article  use the `promo.headlines.seoHeadline` from the Ares payload.

File: data/indonesia/bbc_indonesian_radio/liveradio.json
Comment (ryanmccombe) [reply data/indonesia/bbc_indonesian_radio/liveradio.json:7]: I believe it's called `counterName` in front page, but `pageIdentifier` in articles

https://github.com/bbc/simorgh/blob/82b171addc7c4085ff706fa9483367a1a3eb791d/src/app/containers/ATIAnalytics/params/article/index.jsx#L28

Even though frontpage does call it `counterName`, it gets remapped to `pageIdentifier` before being used in analytics

https://github.com/bbc/simorgh/blob/2c129c231039a63a8c7d5576ca182b8f9133534e/src/app/lib/analyticsUtils/frontpage/index.js#L3

So I think frontpage is the thing that is inconsistent here

The actual data source is temporary in this PR anyway (which is why I only did it for one service) - we acknowledged once the ARES work is complete, there will be some integration updates required (analytics, morph module, `getInitialData`, other places)

File: data/indonesia/bbc_indonesian_radio/liveradio.json
Comment (ryanmccombe) [reply data/indonesia/bbc_indonesian_radio/liveradio.json:8]: Yep - was some back and forward with Iain on whether or not we should have the `promo.headline` object - apparantly it's specific to CPS?

It also doesn't make sense semantically for live radio pages - there is no "headline", although some of our samples now do have this headline prop, some still don't

If necessary, we can fix this once the ares payload is confirmed though

File: data/indonesia/bbc_indonesian_radio/liveradio.json
Comment (pjlee11) [reply data/indonesia/bbc_indonesian_radio/liveradio.json:7]: I believe `pageIdentifier` is the human readable version in the code but Ares will return it as `counterName` for real data so ideally we should have the fixture payloads we use for liveradio match the Ares response for CPS MAPs and on-demand radio

Comment (ryanmccombe) [reply data/indonesia/bbc_indonesian_radio/liveradio.json:7]: Fixture data is a different (blocked) ticket: https://github.com/bbc/simorgh/issues/2754

I can block this PR too behind ARES if preferred though - have asked Iain to provide a sample in the mean time

File: src/app/containers/ATIAnalytics/params/media/index.jsx
Comment (pjlee11) [inline src/app/containers/ATIAnalytics/params/media/index.jsx:17]: This should likely come from the `metadata` rather than being hardcoded, but this can be changed in the future

**Review by pjlee11 — APPROVED** [review]
One minor comment about hardcoding the `contentType` but looks good to me 👍

**Review by j-pendlebury — APPROVED** [review]
Some conflicts need to be resolved ⬇️ 

But otherwise looks good 👍

**Comment by aeroplanejane** [comment]
@ryanmccombe are the pages available on test to check the ATI params?

**Comment by aeroplanejane** [comment]
@ryanmccombe  I don't think this can be tested until our 404 errors are fixed
Had a quick look on: https://www.test.bbc.com/afaanoromoo/bbc_afaanoromoo_radio/liveradio and noted the following errors when viewing with the in-page analytics plugin and the ATI tag inspector 
Screenshot 2019-08-13 at 13 05 22

Screenshot 2019-08-13 at 13 06 58

**Comment by ryanmccombe** [comment]
@aeroplanejane I don't think our test server gets deployed until the code is merged - I believe the way testers test stuff is they have their own local development environment set up, and they pull down the branch and test it locally

**Comment by aeroplanejane** [comment]
Ok, cool - thanks Ryan!

**Comment by ryanmccombe** [comment]
Seems good from the tag inspector
Screen Shot 2019-08-14 at 11 36 11

**Comment by paruchurisilpa** [comment]
Checked http://localhost:7080/afaanoromoo/bbc_afaanoromoo_radio/liveradio..

Screenshot 2019-08-14 at 15 29 45

LGTM..

**Pull request merged**
